plugin/pluginenv/environment.go

```diff
@@ -12,20 +13,26 @@ import (
 type APIProviderFunc func(*plugin.Manifest) (plugin.API, error)
 type SupervisorProviderFunc func(*plugin.BundleInfo) (plugin.Supervisor, error)
 
+type ActivePlugin struct {
+	BundleInfo *plugin.BundleInfo
+	Supervisor plugin.Supervisor
+}
+
 // Environment represents an environment that plugins are discovered and launched in.
```

Comment:
Also, `Environment` isn't safe for concurrent use. I imagine you probably want to just throw a mutex lock around every exported method body.